redis/src/aio/multiplexed_connection.rs

```diff
@@ -185,10 +190,23 @@ where
             }
             ResponseAggregate::Pipeline {
                 expected_response_count,
-                current_response_count,
+                skipped_response_count,
                 buffer,
                 first_err,
             } => {
+                if *skipped_response_count > 0 {
```

Comment:
This is a great change and makes the response handling logic here much less painful to grok. I wonder if some comments added elsewhere to the pipeline code might go a long way in making the pipe vs. transaction logic less confusing.

Reply:
sure, got suggestions or places that you consider confusing? and do you want them as part of this PR?

Reply:
Nah, let's just get this in.